Tests/SwiftParserTest/Linkage.swift

```diff
@@ -0,0 +1,130 @@
+#if canImport(Darwin)
+import Darwin
+import XCTest
+
+final class LinkageTest: XCTestCase {
+  // Assert that SwiftSyntax and SwiftParser do not introduce more link-time
+  // dependencies than are strictly necessary. We want to minimize our link-time
+  // dependencies. If this set changes - in particular, if it grows - consult
+  // a SwiftSyntax maintainer to see if there's a way to avoid adding the
+  // dependency.
+  func testLinkage() throws {
+    for i in 0..<_dyld_image_count() {
+      let name = try XCTUnwrap(_dyld_get_image_name(i))
+      let path = String(cString: name)
+      guard path.hasSuffix("SwiftParserTest") else {
+        continue
+      }
```

Comment:
IIUC we only need this loop to get the path of SwiftParserTest, right? In that case, could we extract the loop to a helper function to make the real test case simpler? I.e. something like

```swift
func getSwiftParserTestBundlePath() -> URL? {
  for i in 0..<_dyld_image_count() {
    let name = try XCTUnwrap(_dyld_get_image_name(i))
    let path = String(cString: name)
    guard path.hasSuffix("SwiftParserTest") else {
      continue
    }
    var baseURL = URL(fileURLWithPath: path)
    while !baseURL.pathComponents.isEmpty, baseURL.pathExtension != "xctest" {
      baseURL = baseURL.deletingLastPathComponent()
    }
    return baseURL
  }
  return nil
}
```

The way it’s currently written, I wonder if we pass the `guard` multiple times.